src/scripts/main.js

```diff
@@ -1,3 +1,178 @@
+}
+
+function handleSortAndUpdate(index, arraySort, directionSort) {
+  const isAscending = directionSort[index];
+
+  const sortedRows = sortRows(index, isAscending, arraySort);
+
+  updateTable(sortedRows);
+
+  directionSort[index] = !isAscending;
+}
+
+tableHeaders.forEach((header, index) => {
+  sortDirection[index] = true;
+
+  header.addEventListener('click', () => {
+    handleSortAndUpdate(index, rowsArray, sortDirection);
+  });
+});
+
+const rows = document.querySelectorAll('tbody tr');
+
+rows.forEach((row) => {
+  row.addEventListener('click', () => {
+    rows.forEach((r) => r.classList.remove('active'));
+
+    row.classList.add('active');
+  });
+});
+
+const formHTML = `
+  <form class="new-employee-form">
+    <label>
+      Name: <input name="name" type="text" data-qa="name" required>
+    </label>
+    <label>
+      Position: <input name="position" type="text" data-qa="position">
+    </label>
+    <label>
+      Age: <input name="age" type="number" data-qa="age" required>
+    </label>
+    <label>
+      Salary: <input name="salary" type="number" data-qa="salary" required>
+    </label>
+    <label>
+      Office:
+      <select name="office" data-qa="office">
+        <option value="Tokyo">Tokyo</option>
+        <option value="Singapore">Singapore</option>
+        <option value="London">London</option>
+        <option value="New York">New York</option>
+        <option value="Edinburgh">Edinburgh</option>
+        <option value="San Francisco">San Francisco</option>
+      </select>
+    </label>
+    <button type="submit">Save to table</button>
+  </form>
+`;
+
+document.body.insertAdjacentHTML('beforeend', formHTML);
+
+const form = document.querySelector('.new-employee-form');
+
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+  validateForm();
+});
+
+function validateForm() {
+  const nameEmployee = form.elements.name.value;
+  const position = form.elements.position.value;
+  const office = form.elements.office.value;
+  const age = parseInt(form.elements.age.value, 10);
+  const salary = parseFloat(form.elements.salary.value);
+
+  const validationError = validateEmployeeDetails(
+    nameEmployee,
+    position,
+    age,
+    salary,
+  );
+
+  if (validationError) {
+    showNotification('Error', validationError, 'error');
+
+    return;
+  }
+
+  addEmployeeAndResetForm(nameEmployee, position, office, age, salary);
+  showNotification('Success', 'Employee added successfully!', 'success');
+}
+
+function validateEmployeeDetails(nameWorker, position, age, salary) {
+  if (nameWorker.length < 4) {
+    return 'Name must be at least 4 characters long.';
+  }
+
+  if (!position) {
+    return 'Please choose a valid position.';
+  }
+
+  if (age < 18 || age > 90) {
+    return 'Age must be between 18 and 90.';
+  }
+
+  if (isNaN(salary) || salary <= 0) {
+    return 'Salary must be a positive number.';
+  }
+
+  return null;
+}
+
+function showNotification(title, message, type) {
```

Comment:
The function `showNotification` is creating a new DOM element for each notification. This could lead to a large number of unnecessary DOM elements if there are many notifications. Consider reusing a single notification element and just changing its text and class.